cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -732,7 +745,7 @@ class ObjectBlock : public PandasBlock {
         }
       }
     } else if (type == Type::STRUCT) {
-      RETURN_NOT_OK(ConvertStruct(data, out_buffer));
+      RETURN_NOT_OK(ConvertStruct(options_, data, out_buffer));
```

Comment:
I had to change most signatures because at some point a Visitor dispatches and the signatures needed to look alike. Maybe I hit one or two functions who didn't deserve it but I figured that this would do much harm